experimental/FTheoryTools/TateModels/constructors.jl

```diff
@@ -0,0 +1,201 @@
+false
+```
+"""
+function global_tate_model(ais::Vector{T}, base::AbstractNormalToricVariety) where {T<:MPolyElem{fmpq}}
+    if length(ais) != 5
+        throw(ArgumentError("We require exactly 5 Tate sections"))
+    end
+    if any(k -> parent(k) != cox_ring(base), ais)
+        throw(ArgumentError("All Tate sections must reside in the Cox ring of the base toric variety"))
+    end
+    toric_ambient_space = _ambient_space_from_base(base)
+    pt = _tate_polynomial(ais, cox_ring(toric_ambient_space))
+    Y4 = closed_subvariety_of_toric_variety(toric_ambient_space, [pt])
+    model = GlobalTateModel(ais[1], ais[2], ais[3], ais[4], ais[5], pt, base, toric_ambient_space, Y4)
+    set_attribute!(model, :base_fully_specified, true)
+    return model
+end
+
+
+################################################
+# 3: Constructors over not fully specified bases
+################################################
+
+@doc Markdown.doc"""
+    global_tate_model(ais::Vector{T}, auxiliary_base_ring::MPolyRing, d::Int) where {T<:MPolyElem{fmpq}}
+
+This method constructs a global Tate model over a base space that is not
+fully specified. The following example exemplifies this approach.
+
+# Examples
+```jldoctest
+julia> auxiliary_base_ring, (a10, a21, a32, a43, a65, w) = QQ["a10", "a21", "a32", "a43", "a65", "w"];
+
+julia> a1 = a10;
+
+julia> a2 = a21 * w;
+
+julia> a3 = a32 * w^2;
+
+julia> a4 = a43 * w^3;
+
+julia> a6 = a65 * w^5;
+
+julia> ais = [a1, a2, a3, a4, a6];
+
+julia> t = global_tate_model(ais, auxiliary_base_ring, 3)
+Global Tate model over a not fully specified base
+
+julia> tate_polynomial(t)
+-a10*x*y*z + a21*w*x^2*z^2 - a32*w^2*y*z^3 + a43*w^3*x*z^4 + a65*w^5*z^6 + x^3 - y^2
+
+julia> toric_base_space(t)
+[ Info: Base space was not fully specified. Returning AUXILIARY base space.
+Normal toric variety
+
+julia> toric_ambient_space(t)
+[ Info: Base space was not fully specified. Returning AUXILIARY ambient space.
+Normal, simplicial toric variety
+
+julia> dim(toric_ambient_space(t))
+5
+```
+"""
+function global_tate_model(ais::Vector{T}, auxiliary_base_ring::MPolyRing, d::Int) where {T<:MPolyElem{fmpq}}
+    if length(ais) != 5
+        throw(ArgumentError("We expect exactly 5 Tate sections"))
+    end
+    if any(k -> parent(k) != auxiliary_base_ring, ais)
+        throw(ArgumentError("All Tate sections must reside in the provided auxiliary base ring"))
+    end
+    if d <= 0
+        throw(ArgumentError("The dimension of the base space must be positive"))
+    end
+    if length([string(k) for k in gens(auxiliary_base_ring)]) < d
+        throw(ArgumentError("We expect at least as many base variables as the desired base dimension"))
+    end
+
+    # convert Tate sections into polynomials of the auxiliary base
+    auxiliary_base_space = _auxiliary_base_space([string(k) for k in gens(auxiliary_base_ring)], d)
+    S = cox_ring(auxiliary_base_space)
+    ring_map = hom(auxiliary_base_ring, S, [gens(S)[i] for i in 1:length(gens(S))])
```

Comment:
Why not just this?
```suggestion
    ring_map = hom(auxiliary_base_ring, S, gens(S))
```
and indeed, I think even this might work (but I might be wrong, please verify
```suggestion
    ring_map = hom(auxiliary_base_ring, S)
```

Reply:
I have tried to optimize the construction of all ring maps in FTheoryTools.

I received an error from your last suggestion, i.e. without `gens(S)`, but the suggestion above works quite well. In fact, in most places we need a subset of `gens(S)` for these maps. Unless I am mistaken, there is only instance where once could hope to get away without specifying the images. So maybe not a big drawback.